src/components/graph/menus/network-modification-node-editor.tsx

```diff
@@ -249,72 +255,66 @@ const NetworkModificationNodeEditor = () => {
                 editData={editData}
                 isUpdate={isUpdate}
                 editDataFetchStatus={editDataFetchStatus}
-                {...props}
             />
         );
     }
 
-    function adapt(Dialog, ...augmenters) {
-        const nprops = augmenters.reduce((pv, cv) => cv(pv), {});
-        return withDefaultParams(Dialog, nprops);
-    }
-
-    const menuDefinition = [
+    const menuDefinition: MenuDefinition[] = [
         {
             id: 'CREATE',
             label: 'menu.create',
             subItems: [
                 {
                     id: MODIFICATION_TYPES.SUBSTATION_CREATION.type,
                     label: 'SUBSTATION',
-                    action: () => adapt(SubstationCreationDialog),
+                    action: () => withDefaultParams(SubstationCreationDialog),
                 },
                 {
                     id: MODIFICATION_TYPES.VOLTAGE_LEVEL_CREATION.type,
                     label: 'VOLTAGE_LEVEL',
-                    action: () => adapt(VoltageLevelCreationDialog),
+                    action: () => withDefaultParams(VoltageLevelCreationDialog),
                 },
                 {
                     id: MODIFICATION_TYPES.LINE_CREATION.type,
                     label: 'LINE',
-                    action: () => adapt(LineCreationDialog),
+                    action: () => withDefaultParams(LineCreationDialog),
                 },
                 {
                     id: MODIFICATION_TYPES.TWO_WINDINGS_TRANSFORMER_CREATION.type,
                     label: 'TWO_WINDINGS_TRANSFORMER',
-                    action: () => adapt(TwoWindingsTransformerCreationDialog),
+                    action: () => withDefaultParams(TwoWindingsTransformerCreationDialog),
                 },
                 {
                     id: 'GENERATOR_CREATION',
```

Comment:
why not use the enum?

Reply:
Do you think it has to be done here ?
Migrating to TS is already enough in my opinion